src/transports/janus_websockets.c

```diff
@@ -1287,35 +1287,49 @@ static int janus_websockets_common_callback(
 			/* Refresh the lws connection validity (avoid sending a ping) */
 			lws_validity_confirmed(ws_client->wsi);
 #endif
+			size_t incoming_length;
 			/* Is this a new message, or part of a fragmented one? */
 			const size_t remaining = lws_remaining_packet_payload(wsi);
 			if(ws_client->incoming == NULL) {
 				JANUS_LOG(LOG_HUGE, "[%s-%p] First fragment: %zu bytes, %zu remaining\n", log_prefix, wsi, len, remaining);
 				ws_client->incoming = g_malloc(len+1);
 				memcpy(ws_client->incoming, in, len);
-				ws_client->incoming[len] = '\0';
+				incoming_length = len;
+				ws_client->incoming[incoming_length] = '\0';
 				JANUS_LOG(LOG_HUGE, "%s\n", ws_client->incoming);
 			} else {
 				size_t offset = strlen(ws_client->incoming);
 				JANUS_LOG(LOG_HUGE, "[%s-%p] Appending fragment: offset %zu, %zu bytes, %zu remaining\n", log_prefix, wsi, offset, len, remaining);
 				ws_client->incoming = g_realloc(ws_client->incoming, offset+len+1);
 				memcpy(ws_client->incoming+offset, in, len);
-				ws_client->incoming[offset+len] = '\0';
+				incoming_length = offset+len;
+				ws_client->incoming[incoming_length] = '\0';
 				JANUS_LOG(LOG_HUGE, "%s\n", ws_client->incoming+offset);
 			}
 			if(remaining > 0 || !lws_is_final_fragment(wsi)) {
 				/* Still waiting for some more fragments */
 				JANUS_LOG(LOG_HUGE, "[%s-%p] Waiting for more fragments\n", log_prefix, wsi);
 				return 0;
 			}
-			JANUS_LOG(LOG_HUGE, "[%s-%p] Done, parsing message: %zu bytes\n", log_prefix, wsi, strlen(ws_client->incoming));
+			JANUS_LOG(LOG_HUGE, "[%s-%p] Done, parsing message: %zu bytes\n", log_prefix, wsi, incoming_length);
 			/* If we got here, the message is complete: parse the JSON payload */
-			json_error_t error;
-			json_t *root = json_loads(ws_client->incoming, 0, &error);
+			const char* request_curr = ws_client->incoming;
+			const char* request_end = ws_client->incoming + incoming_length;
+			do { /* load all json objects from the message */
+				json_error_t error;
+				json_t *root = json_loads(request_curr, JSON_DISABLE_EOF_CHECK, &error);
```

Comment:
This is a cool flag I was unaware of, but I'm wondering if it may cause issues in some cases, e.g., with malformed messages (think of one with two closing brackets, for instance). This may cause the decoder to accept part of the input as valid when it really isn't, where right now we have some validation on incoming messages. I also see this is only available starting from version 2.1 of Jansson, so in case this is accepted it means we'll need to explicitly add this requirement to `configure.ac` as well, or it will break on instances using older versions of the library (I'm not sure how common those are).

Reply:
Looks like configure.ac already specifies 2.5 for Jansson - so no work on that front yeah?

Reply:
I see what you mean. I've pushed updated code that parses all the messages before processing any and fails in entirety if json is not well formed until EOF. I'm gonna pull it into my fork for testing, so I expect there might be another update to this.

Reply:
With the udpates, with two closing brackes I now get this error:

```
Returning Janus API error 454 (Invalid Janus API request - <string>(1,1): '[' or '{' expected near '}')
```

and the message before the extra closing is ignored.

Reply:
Testing from my fork this is working well - I'm seeing multiple requests handled and errors for all if there's any parsing problems in the set.